paywall/src/components/content/CheckoutContent.tsx

```diff
@@ -69,12 +92,22 @@ export default function CheckoutContent() {
     allowClosingCheckout = true
   }
   // hide the checkout iframe
-  const hideCheckout = () => {
+  const hideCheckout = useCallback(() => {
     postMessage({
       type: POST_MESSAGE_DISMISS_CHECKOUT,
       payload: undefined, // this must be set to trigger a response in unlock.min.js
     })
-  }
+  }, [postMessage])
+
+  // get a list of locks that have valid keys
+  const activeLocks = Object.keys(locks as Locks).filter(
+    (lockAddress: string) => locks[lockAddress].key.status === 'valid'
+  )
+  useEffect(() => {
+    if (activeLocks.length && !userInitiatedPurchase) {
```

Comment:
this is the new part: auto-hide the Checkout UI if any locks are valid and we did not initiate the purchase in this session